src/cli/cli_dataset.cpp

```diff
@@ -485,6 +486,7 @@ otError Dataset::ProcessNetworkName(uint8_t aArgsLength, char *aArgs[])
         size_t length;
 
         VerifyOrExit((length = strlen(aArgs[0])) <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
+        VerifyOrExit(ValidateUtf8(aArgs[0]), error = OT_ERROR_INVALID_ARGS);
```

Comment:
Does this cover the case where dataset is set using hex format?

https://github.com/openthread/openthread/blob/master/src/cli/README_DATASET.md#set

Reply:
No. The current implementation of hex set doesn't do any check. It succeeds even if the data is just some random binary data. I think it's good to add the parsing and verification, but that also means some runtime overhead.

Reply:
Added checks for `dataset set`.